lib/Parse/ParseDecl.cpp

```diff
@@ -6748,6 +6748,17 @@ Parser::parseDeclVar(ParseDeclOptions Flags,
     // If we syntactically match the second decl-var production, with a
     // var-get-set clause, parse the var-get-set clause.
     if (Tok.is(tok::l_brace)) {
+
+      // Skip parsing the var-get-set clause if '{' is at start of line
+      // and next token is not 'didSet' or 'willSet'. Parsing as 'do'
+      // statement gives useful errors for missing 'do' before brace.
+      // See SR-14836.
+      Token NextToken = peekToken();
+      if (Tok.isAtStartOfLine() && !NextToken.isContextualKeyword("didSet") &&
+          !NextToken.isContextualKeyword("willSet")) {
+        break;
```

Comment:
`break` here means it skips [the rest of this block](https://github.com/apple/swift/blob/ca26a1a3fe24ce636c5e97dc7701c5b6669caf42/lib/Parse/ParseDecl.cpp#L6760-L6780). I don't think that's desirable.